src/stream/future.rs

```diff
@@ -14,6 +14,46 @@ pub fn new<S: Stream>(s: S) -> StreamFuture<S> {
     StreamFuture { stream: Some(s) }
 }
 
+impl<S> StreamFuture<S> {
+    /// Acquires a reference to the underlying stream that this combinator is
+    /// pulling from.
+    ///
+    /// This method returns an `Option` to account for the fact that `StreamFuture`'s
+    /// implementation of `Future::poll` consumes the underlying stream during polling 
+    /// in order to return it to the caller of `Future::poll` if the stream yieleded
```

Comment:
_yielded_ 😉

Reply:
PR for fix: https://github.com/rust-lang-nursery/futures-rs/pull/723

Reply:
And thanks. :)